### PR TITLE
add potential exception handle & open session_option to optimize

### PR DESCRIPTION
**Pull request by yanwei-gr**

### CODE REVIEW

File: ODLA/platforms/odla_popart/popart_config.cc [hunk 1]
```diff
@@ -56,14 +56,18 @@ void PopartConfig::use_default() {
       }\n";
 }
 
-void PopartConfig::load_config(const char* file_path) {
+odla_status PopartConfig::load_config(const char* file_path) {
+  odla_status ret_value = ODLA_SUCCESS;
```
Comment (jackzipu) [inline ODLA/platforms/odla_popart/popart_config.cc:60]: Seemed we don't need this variable?

File: ODLA/platforms/odla_popart/popart_config.cc [hunk 2]
```diff
@@ -200,6 +213,7 @@ bool PopartConfig::get_pipeline_setting(const std::string& node_name,
     pipeline_stage = default_setting_iter->second[1];
     return true;
   }
+
   throw std::runtime_error(
```
Comment (jackzipu) [inline ODLA/platforms/odla_popart/popart_config.cc:217]: Why we kept this throw?

**Review by jackzipu — COMMENTED** [review]
I've done the review and added comments.  Thanks.

File: ODLA/platforms/odla_popart/popart_config.cc (hunk 2 above)
Comment (yanwei-gr) [reply ODLA/platforms/odla_popart/popart_config.cc:217]: exception is caught outside. in builder::createFromOnnx

**Review by jackzipu — APPROVED** [review]
Seemed good to me.

**Review by jackzipu — APPROVED** [review]
I'm good for the changes.  Thanks.

**Review by weimingzha0 — APPROVED** [review]

**Review by littlefatfat — APPROVED** [review]

**Pull request merged**